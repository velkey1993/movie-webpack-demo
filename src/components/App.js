import React from 'react';
import '../styles/App.css';
import Header from "./Header";
import Footer from "./Footer";
import Main from "./Main";
import Sidebar from "./Sidebar";
import Navbar from "./Navbar";

function App({imageId, headerText}) {
    return (
        <div id={imageId}>
            <Navbar/>
            <Header text={headerText}/>
            <Main text={"Main"}/>
            <Sidebar text={"Sidebar"}/>
            <Footer text={"Footer"}/>
        </div>
    );
}

export default App;
