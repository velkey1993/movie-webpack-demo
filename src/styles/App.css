h1 {
    font-family: Arial, serif;
    color: blue;
}

#bird-image {
    background: url('../assets/bird.jpg');
    width: 220px;
    height: 329px;
}
