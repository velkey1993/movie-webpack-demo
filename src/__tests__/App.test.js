import React from 'react';
import renderer from "react-test-renderer";
import App from '../components/App';
import {test} from "@jest/globals";

test('renders app component', () => {
    const app = renderer.create(<App imageId={"bird-image"}/>);
    const appRendered = app.toJSON();
    expect(appRendered).toMatchSnapshot();
    expect(app.root._fiber.memoizedProps.imageId).toBe("bird-image");
});
