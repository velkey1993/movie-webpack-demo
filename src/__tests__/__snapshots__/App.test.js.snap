// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component 1`] = `
<div
  id="bird-image"
>
  <ul>
    <li>
      <a
        href="index.html"
      >
        Index
      </a>
    </li>
    <li>
      <a
        href="contact.html"
      >
        Contact
      </a>
    </li>
    <li>
      <a
        href="about.html"
      >
        About
      </a>
    </li>
  </ul>
  <header>
    <h1 />
  </header>
  <main>
    <h1>
      Main
    </h1>
  </main>
  <aside>
    <h1>
      Sidebar
    </h1>
  </aside>
  <footer>
    <h1>
      Footer
    </h1>
  </footer>
</div>
`;
